src/main/java/uk/gov/hmcts/reform/userprofileapi/controller/UserProfileController.java

```diff
@@ -370,10 +398,55 @@ public ResponseEntity updateUserProfile(@Valid @RequestBody UpdateUserProfileDat
 
         boolean showDeletedBoolean = UserProfileValidator.validateAndReturnBooleanForParam(showDeleted);
         boolean rolesRequiredBoolean = UserProfileValidator.validateAndReturnBooleanForParam(rolesRequired);
-        UserProfileValidator.validateUserIds(userProfileDataRequest);
+        validateUserIds(userProfileDataRequest);
         UserProfileDataResponse userProfileDataResponse =
                 userProfileService.retrieveWithRoles(new UserProfileIdentifier(IdentifierName.UUID_LIST, userProfileDataRequest.getUserIds()), showDeletedBoolean, rolesRequiredBoolean);
         return ResponseEntity.status(HttpStatus.OK).body(userProfileDataResponse);
 
     }
+
+    @ApiOperation(value = "Delete an User Profiles",
+            authorizations = {
+                    @Authorization(value = "ServiceAuthorization"),
+                    @Authorization(value = "Authorization")
+            })
+    @ApiResponses({
+            @ApiResponse(
+                    code = 204,
+                    message = "User Profiles deleted successfully",
+                    response = UserProfilesDeletionResponse.class
+            ),
+            @ApiResponse(
+                    code = 400,
+                    message = "An invalid request has been provided"
+            ),
+            @ApiResponse(
+                    code = 401,
+                    message = "Unauthorized Error : The requested resource is restricted and requires authentication"
+            ),
+            @ApiResponse(
+                    code = 403,
+                    message = "Forbidden Error: Access denied"
+            ),
+            @ApiResponse(
+                    code = 500,
+                    message = "User Profiles not deleted successfully"
+            )
+    })
+
+    @DeleteMapping(
+            consumes = APPLICATION_JSON_VALUE,
+            produces = APPLICATION_JSON_VALUE
+    )
+    @ResponseStatus(value = HttpStatus.NO_CONTENT)
+    @ResponseBody
+    public ResponseEntity<UserProfilesDeletionResponse> deleteUserProfiles(@Valid @RequestBody UserProfileDataRequest userProfilesDeletionDataReq) {
+
+        UserProfilesDeletionResponse resource = null;
+        validateUserIds(userProfilesDeletionDataReq);
+        resource = userProfileService.delete(new UserProfilesDeletionData(userProfilesDeletionDataReq.getUserIds()));
+        return ResponseEntity.status(resource.getStatusCode()).body(resource);
```

Comment:
No need to create new request class UserProfilesDeletionData since list of userIDs from UserProfileDataRequest can be directly sent to service and sent same list to deleter

Reply:
done